src/main/java/build/buf/protovalidate/internal/evaluator/ObjectValue.java

```diff
@@ -50,17 +49,25 @@ public final class ObjectValue implements Value {
     this.value = value;
   }
 
-  @Nullable
   @Override
-  public Message messageValue() {
-    if (fieldDescriptor.getType() == Descriptors.FieldDescriptor.Type.MESSAGE) {
-      return (Message) value;
+  public MessageLike messageValue() {
+    return new ProtobufMessageLike((Message) value);
```

Comment:
Why is this behavior changing to not return null if the underlying field isn't a message?

Reply:
I think what's happening is protovalidate always calls this correctly - only when the underlying field is a message. Just like the Set NPE you mention above, it's not a bad idea to defend against accidental bugs.